src/classLibrary/ParsingStrategies/IfElseResolver.ts

```diff
@@ -0,0 +1,210 @@
+import {Utility} from "../Utility/Utility";
+import {FileSystemInstance, 
+	FileContent,
+	VirtualFileSystemInstance,
+	IParsingStrategy,
+	Syntax,
+	CyanSafe,
+	} from "../interfaces/interfaces";
+
+class IfElseResolver implements IParsingStrategy {
+
+	private util: Utility;
+	
+	constructor(util: Utility) {
+		this.util = util;
+	}
+	
+	Count(cyan: CyanSafe, virtualFiles: VirtualFileSystemInstance[]): Map<string, number> {
+    	const flags: string[] = this.util.FlattenObject(cyan.flags).Keys();
+        const result: Map<string, number> = new Map<string, number>();
+        const syntaxes: Syntax[] = cyan.syntax;
+
+        virtualFiles.Each((virtualFile: VirtualFileSystemInstance) => {
+            flags.Each((flag: string) => {
+                const allPossibleIfs = this.ModifyIfWithAllSyntax(flag, syntaxes).concat(this.ModifyInverseIfWithAllSyntax(flag, syntaxes));
+                allPossibleIfs.Each((key: string) => {
+                    const count = this.CountKeyInVFS(key, virtualFile)
+									.AtMax(this.CountKeyInVFS(this.ConvertIfToEndKeyword(key), virtualFile));
```

Comment:
Is this line just ensuring that the number of `if~smth~` is not greater than `end~smth`?

Reply:
the overall count would would at max be the count of the end~smth so yep